File: GitToolBox/src/main/java/zielu/gittoolbox/ResBundle.java
package zielu.gittoolbox;

import com.intellij.BundleBase;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.PropertyKey;
import zielu.intellij.util.ZBundleHolder;

public final class ResBundle {
  @NonNls
  private static final String BUNDLE_NAME = "zielu.gittoolbox.ResourceBundle";
  private static final ZBundleHolder BUNDLE_HOLDER = new ZBundleHolder(BUNDLE_NAME);

  private ResBundle() {
    throw new IllegalStateException();
  }

  @NotNull
  public static String message(@NotNull @PropertyKey(resourceBundle = BUNDLE_NAME) String key, Object... params) {
    return BundleBase.message(BUNDLE_HOLDER.getBundle(), key, params);
  }

  public static String na() {
    return message("common.na");
  }

  public static String naLabel() {
    return message("common.na.label");
  }

  public static String on() {
    return message("common.on");
  }

  public static String disabled() {
    return message("common.disabled");
  }

  public static String example() {
    return message("common.example.label");
  }
}

File: GitToolBox/src/main/kotlin/zielu/gittoolbox/ResBundle.kt
package zielu.gittoolbox

import com.intellij.DynamicBundle
import org.jetbrains.annotations.NonNls
import org.jetbrains.annotations.PropertyKey

internal object ResBundle : DynamicBundle(BUNDLE_NAME) {
  @JvmStatic
  fun message(@PropertyKey(resourceBundle = BUNDLE_NAME) key: String, vararg params: Any?): String {
    return getMessage(key, *params)
  }

  @JvmStatic
  fun na(): String {
    return message("common.na")
  }

  @JvmStatic
  fun on(): String {
    return message("common.on")
  }

  @JvmStatic
  fun disabled(): String {
    return message("common.disabled")
  }
}

@NonNls
private const val BUNDLE_NAME = "zielu.gittoolbox.ResourceBundle"
